fix server instance numbers for config

Berthalamew opened the pull request:
- dedis will now use local configs by default
- dedis that have an instance name supplied will use h2serverconfig appended by the instance parameter supplied
- dedis will now use the instance name appended to the log instance folder
- add unicode helper functions
- replace Memory::IsDedicatedServer with shell_is_dedicated_server
- make server use our rewritten shell_initialize function
- remove RunLoop
- move crc table in Util header to cpp
- add custom setting struct in playlist loader
- remove experimental rendering enum
- remove filesys
- fix shadow and water graphics options defaulting to low
- don't initialize console commands on dedis

Berthalamew committed:
add shell_is_dedicated_server and replace Memory::IsDedicatedServer

Berthalamew committed:
fix server instance numbers for config

- make server use our rewritten shell_initialize function
- remove RunLoop
- move crc table in Util header to cpp
- add custom setting struct in playlist loader
- remove experimental rendering enum

Berthalamew committed:
reimplement main window creation function

- change how rdp check is disabled (we don't receive WM_WTSSESSION_CHANGE messages anymore)
- re-enables resizing during startup and fixes resizing being disabled when changing resolutions

Berthalamew committed:
adjust sock string functions in xlive

- use string pointers instead of copying/constructing the string constantly

Berthalamew committed:
adjust nointro arg and config option usage

- if the intro is already disabled via command line flag don't try and set it via the config option

nukeulater commented on xlive/Blam/Engine/shell/shell.cpp:
```diff
@@ -80,107 +96,79 @@ const char* shell_get_version(void)
-	if (cmd_line_args && arg_count > 1) {
-		for (int i = 1; i < arg_count; i++) {
-			wchar_t* cmd_line_arg = cmd_line_args[i];
+	SYSTEM_DEBUG_MEMORY(cseries_initialize());
```
speaking of `SYSTEM_DEBUG_MEMORY`, the macro should also have a "before" initialization message, and all of these initialization calls should be 1 liners:

```
SYSTEM_DEBUG_MEMORY(real_math_initialize());
```

not:

```
real_math_initialize()
SYSTEM_DEBUG_MEMORY(real_math_initialize());
```

where SYSTEM_DEBUG_MEMORY is different depending on the compile configuration used

Berthalamew replied:
I think this is actually how the macro was originally, since `runtime_state_initialize` is called after `cseries_initialize` but the debug memory internal call is done after `runtime_state_initialize`

Berthalamew replied:
Same thing with `animation_cache_new`. There's no animation cache on halo 2 vista but a memory internal call is done after it would of been there like on xbox.

nukeulater replied:
nvm, there's no "before" message, I miscalled that

Berthalamew replied:
There does seem to be a before macro as well (i.e. before XNetStartup), they're just not used in that shell code.

Berthalamew committed:
add unicode functions

- update function names

Berthalamew committed:
add error category enum and constants

Berthalamew committed:
adjust config code

- if a local config exists it will use it by default
- dedis will now use local configs by default
- dedis that have an instance name supplied will use h2serverconfig appended by the instance parameter supplied

Berthalamew committed:
move cartographer urls to endpoints file

Berthalamew committed:
adjust code to set the process path

- remove filesys
- remove GetWidePathFromFullWideFilename

Berthalamew committed:
fix instance number logging bug

- instance number was being set after logging was initialized
- move logging creation to it's own function
- adjust logic in InitLocalAppData (remove Local Settings code)

Berthalamew committed:
fix shadow and water graphics options defaulting to low

Berthalamew committed:
fix logging to use the server instance name

Berthalamew committed:
misc changes

- don't initialize console commands on the dedi
- fix null string message

Berthalamew committed:
hook up portable logs properly

Berthalamew committed:
fix appdata folder creation logic
